src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java

```diff
@@ -23,5 +23,44 @@ public static void main(String[] args) {
                 new UserMeal(LocalDateTime.of(2015, Month.MAY, 31, 13, 0), "Обед", 500),
                 new UserMeal(LocalDateTime.of(2015, Month.MAY, 31, 20, 0), "Ужин", 510)
         );
+
+        getFilteredWithExceeded(mealList, LocalTime.of(7, 0), LocalTime.of(12, 0), 2000);
+//        .toLocalDate();
+//        .toLocalTime();
+
+    }
+
+    public static List<UserMealWithExceed> getFilteredWithExceeded(List<UserMeal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
+        //System.out.println("Need to return List");
+        Map<LocalDate, Integer> mapCaloriesPerDay = new HashMap<>();
+        for (UserMeal meal : mealList
+        ) {
+            LocalDate date = meal.getDateTime().toLocalDate();
+            Integer newCalories = meal.getCalories();
+            mapCaloriesPerDay.merge(date, newCalories, Integer::sum);
+        }
+
+        List<UserMealWithExceed> mealWithExceedList = new ArrayList<>();
+        for (UserMeal meal : mealList
+        ) {
+            if (TimeUtil.isBetween(meal.getDateTime().toLocalTime(), startTime, endTime)) {
+                if (mapCaloriesPerDay.get(meal.getDateTime().toLocalDate()) > caloriesPerDay) {
+                    mealWithExceedList.add(new UserMealWithExceed(meal.getDateTime(), meal.getDescription(),
+                            meal.getCalories(), true));
```

Comment:
Use the following format when you need to carry the parameters: 
mealWithExceedList.add(
new UserMealWithExceed(meal.getDateTime(), meal.getDescription(),meal.getCalories(), true)
);

Reply:
Should I use tab in this way?
mealWithExceedList.add(
       new UserMealWithExceed(meal.getDateTime(), meal.getDescription(), meal.getCalories(), true));